src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java

```diff
@@ -322,6 +332,38 @@ public UUID getStudyRootNodeUuid(UUID studyId) {
         return nodesRepository.findByStudyIdAndType(studyId, NodeType.ROOT).orElseThrow(() -> new StudyException(ELEMENT_NOT_FOUND)).getIdNode();
     }
 
+    @Transactional(readOnly = true)
+    public void assertNodeNameNotExist(UUID studyUuid, String nodeName) {
+        if (isNodeNameExists(studyUuid, nodeName)) {
+            throw new StudyException(NODE_NAME_ALREADY_EXIST);
+        }
+    }
+
+    @Transactional(readOnly = true)
+    public boolean isNodeNameExists(UUID studyUuid, String nodeName) {
+        return ROOT_NODE_NAME.equals(nodeName) || !networkModificationNodeInfoRepository.findAllByNodeStudyIdAndName(studyUuid, nodeName).isEmpty();
+    }
+
+    private List<String> getStudyNodesNames(UUID studyUuid) {
+        return networkModificationNodeInfoRepository.findAllByNodeStudyId(studyUuid)
+                .stream()
+                .map(node -> node.getName())
+                .collect(Collectors.toList());
+    }
+
+    public String getUniqueNodeName(UUID studyUuid) {
```

Comment:
add @Transactional(readOnly = true)
and move code getStudyNodesNames inside this method

Reply:
ok, done